### ISSUE TITLE
AndroidClientEngine set a Content-Lenght: 2 on a GET request with an EmptyContent as body

### ISSUE THREAD

**Issue by kwiky** [ktorio/ktor#1900]
**Ktor Version and Engine Used (client or server and name)**
1.3.2

**Describe the bug**
1. On a `GET` request, the body is an `EmptyContent`
2. The `JsonFeature` with a `KotlinxSerializer` serialize this `EmptyContent` as `{}` [Code Here](https://github.com/ktorio/ktor/blob/7eeb56ccf17c5cbc7a60013ac575d2f72822dcf7/ktor-client/ktor-client-features/ktor-client-json/common/src/io/ktor/client/features/json/JsonFeature.kt#L116)
3. The `AndroidClientEngine` calculate `Content-Length` based on this `{}` and set it to `2` [Code Here](https://github.com/ktorio/ktor/blob/c03d6b2977d9bf54e608d5fb44c8040d8219ae41/ktor-client/ktor-client-android/jvm/src/io/ktor/client/engine/android/AndroidClientEngine.kt#L44)
4. Then the `GET` request is executed without body but with a `Content-Lenght: 2` header

**Expected behavior**
 `GET` request executed without body and with a `Content-Lenght: 0` header

Maybe, [this PR](https://github.com/ktorio/ktor/pull/1644/files) could resolve this issue ?

**Comment by kwiky** [comment]
I think i found a solution
I have to set the content-type only when necessary :
```
        if (method == HttpMethod.Post || method == HttpMethod.Put || method == HttpMethod.Patch) {
            contentType(ContentType.Application.Json)
        }
```

**Comment by stefannegele** [comment]
This is not specific to Android. I´m experiencing the same with the curl engine.

**Comment by e5l** [comment]
IHi @kwiky, @stefannegele. Thanks for the report.
Right now, the `EmptyContent` and `Unit` works the same way: they produce the `{}` string. Could you tell me why do you need to set `EmptyContent` with the specific ContentType?

**Comment by micutad** [comment]
Experienced the same issue. Debug it deep until this code:
`is EmptyContent -> feature.serializer.write(Unit, contentType)`
inside JsonFeature
on first look this seems like mistake as Empty content definitely isnt `{}` and it could be fixed with 
`is EmptyContent -> EmptyContent`
as EmptyContent is already of OutgoingContent.NoContent type

**Comment by stefannegele** [comment]
@e5l sorry for the late answer. The problem was that produced body was empty (not "{}") but the content length was still 2. The API I was working against then produces the error, because it was trying to read a content with this length. Wit a POST, it did. I guess, it's not an issue with most systems, because they do not try to read anything in a GET.

**Comment by oleg-larshin** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/KTOR-361">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.

**Comment by e5l** [comment]
Fixed in `1.4.1`
